lib/ansible/modules/cloud/docker/docker_swarm_service.py

```diff
@@ -622,13 +622,19 @@
         type: int
   state:
     description:
-      - Service state.
+      - I(absent) - A service matching the specified name will be removed and have its tasks stopped.
+      - I(present) - Asserts the existence of a service matching the name and provided configuration parameters.
+        Unspecified configuration parameters will be set to docker defaults.
+      - I(updated) - Will update a service matching the name with provided configuration parameters.
+        Unspecified configuration parameters already set on the service will be untouched. This option only
```

Comment:
I'm not sure whether this will really work. From how this is implemented on docker-py level, it only considers whole subobjects, like `rollback_config`. So if you leave `rollback.delay` away, but specify other `rollback.xxx` options, the value of `rollback.delay` will not be kept, but set to the default value, because the module passes `rollback_config` to docker-py.

If this behavior is intended, then the documentation must be a lot more explicit about this :)

Reply:
I continued this discussion as a comment as it really touches this whole PR.